### ISSUE TITLE
[Visual Requirements – Make Code - Create GitHub Repository] Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds

### ISSUE THREAD

**Issue by Shalini3457** [microsoft/pxt-microbit#3759]
User Experience:
Visual impaired user would find difficulty to see the control or element if that controls have insufficient color contrast as they won't be able to take on that controls and he/she would miss the functionality of such control or element.

Note: User credentials should NOT be included in the bug.

Repro Steps:

Launch application URL
Home page will open.
Navigate to "Import" button and select it.
Navigate through the controls appearing under "Import" dialog.
Navigate to "Your GitHub Repo" option and Select it.
Navigate and select "Sign In" control under Sign in with GitHub and navigate to all the controls of dialogue "Clone or create your own GitHub repo
Navigate and select "Create New" control under "Clone or create your own GitHub repo" and navigate to all the controls of dialogue "Create GitHub Repository
Verify the issue.

Actual Result:
Luminosity ratio is less than 4.5:1 i.e. 3.22 for text control(Repository names must be less than 64 characters and cannot include spaces or special characters.)

Expected Result:
Luminosity ratio should be more than and equal to 4.5:1 for text control (Repository names must be less than 64 characters and cannot include spaces or special characters.)

Issue: Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds (color-contrast - https://dequeuniversity.com/rules/axe/4.0/color-contrast?application=msftAI)

Target application: Microsoft MakeCode for micro:bit - https://makecode.microbit.org/#

Element path: .yellow

Snippet: <div class="ui yellow message">Repository names must be less than 64 characters and cannot include spaces or special characters.</div>

How to fix:
Fix any of the following:
 Element has insufficient color contrast of 3.22 (foreground color: #b58105, background color: #fff8db, font size: 12.0pt (16px), font weight: normal). Expected contrast ratio of 4.5:1

Environment: Microsoft Edge version 88.0.692.0

Note:
Same issue exists on Firefox & Chrome.

WCAG Reference:
[WCAG 1.4.3 - Non-text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)

Test Environments:
Version: 20H2 (OS Build: 19042.508)
Browser: Edge chromium DEV Build (Version 87.0.654.0), Firefox(75.0), Chrome
Tool: Accessibility insights for web
Screen Reader: Narrator, NVDA, JAWS

**Comment by Shalini3457** [comment]
Labels:
#Accessibility; #A11ySev2; #HCL; #HCL-MakeCode-New; #WCAG1.4.3; #Win10-Edge(Chromium);

![Uploading WCAG1.4.3.png…]()
